src/amo/components/AddonsByAuthorsCard/index.js

```diff
@@ -205,6 +209,13 @@ export class AddonsByAuthorsCardBase extends React.Component<InternalProps> {
       type,
     } = this.props;
 
+    invariant(errorHandler, 'errorHandler is required');
+
+    if (errorHandler.hasError()) {
```

Comment:
This seems like the wrong thing to do because it is swallowing any error that may have been produced by the sagas in this component. If I remove this if statement, the original bug you set out to fix is still fixed: I see the 404 page.

However, swallowing errors looks like an unrelated bug that you may as well fix. I think you can fix it like this:

```diff
diff --git a/src/amo/components/AddonsByAuthorsCard/index.js b/src/amo/components/AddonsByAuthorsCard/index.js
index e9c395e43..74873c400 100644
--- a/src/amo/components/AddonsByAuthorsCard/index.js
+++ b/src/amo/components/AddonsByAuthorsCard/index.js
@@ -211,11 +211,6 @@ export class AddonsByAuthorsCardBase extends React.Component<InternalProps> {
 
     invariant(errorHandler, 'errorHandler is required');
 
-    if (errorHandler.hasError()) {
-      log.warn('Captured API Error:', errorHandler.capturedError);
-      return null;
-    }
-
     if (!loading && (!addons || !addons.length)) {
       return null;
     }
@@ -341,17 +336,20 @@ export class AddonsByAuthorsCardBase extends React.Component<InternalProps> {
     }
 
     return (
-      <AddonsCard
-        addons={addons}
-        className={classnames}
-        footer={paginator}
-        header={header}
-        loading={loading}
-        placeholderCount={numberOfAddons}
-        showMetadata
-        showSummary={showSummary}
-        type={type}
-      />
+      <React.Fragment>
+        {errorHandler.renderErrorIfPresent()}
+        <AddonsCard
+          addons={addons}
+          className={classnames}
+          footer={paginator}
+          header={header}
+          loading={loading}
+          placeholderCount={numberOfAddons}
+          showMetadata
+          showSummary={showSummary}
+          type={type}
+        />
+      </React.Fragment>
     );
   }
 }

```

Reply:
> I also decided to hide the AddonsByAuthorsCard when an error occurs,
because it is better than being stuck in a loading state.

Ah, I just read this part in your description.

In this case, you should switch it in an if statement, like:

```js
render() {
  if (errorHandler.hasError()) {
    return errorHandler.renderHoweverItsCalled();
  } else {
    return <AddonsCard ...>
  }
}
```
I don't think you should hide the error, though.

Reply:
>If I remove this if statement, the original bug you set out to fix is still fixed: I see the 404 page.

Yes, I added this after..